### PR TITLE
D2k map pool update - november 2023

### PR DESCRIPTION
**Pull request by Porenutak**
First commit adds new cliff passages that these (and manny other) maps use. All new tiles was created by me and are free to use.
Second commit update map pool:
new maps: worm_graveyard.oramap
update maps: red-chasm.oramap, death-depths.oramap, the-duell.oramap
removed maps: cliffmaze

Updated maps changelog: equal distance from spice fields for each player, equal amount of spice for each player and minor balance adjustments.

Ps. Worm graveyard use repairble passages by enginear.

### CODE REVIEW

**Comment by PunkPun** [comment]
I dunno why but this PR needs a rebase

**Comment by Porenutak** [comment]
I rebase on my local branch, but when I try to pull it on github I get this error:
Screenshot (243)
I also cant Squash rebase commits:
Screenshot (245)

Can someone help me with this?
Im noob with Github and not sure whats wrong.

**Review by pchote — COMMENTED** [review]
A few nits:
* worm_graveyard.oramap contains custom rules, so should be commited as an unpacked map.
* Please rename NewTiles.shp to something more descriptive.
* Please keep other filenames lower case (worm_graveyard's custom yaml uses a mix of CamelCase and lower case)

**Comment by Porenutak** [comment]
Updated according to your suggestions.
NewTiles.shp renamed to custompassagetiles.shp

File: mods/d2k/maps/worm-graveyard/destroyable-tiles.yaml [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+engineer:
+	Captures@cliff:
+		CaptureTypes: cliff
+		ConsumedByCapture: false
+		CaptureDelay: 150
+		ValidRelationships: Enemy, Neutral, Ally
+
+
+
```
Comment (PunkPun) [inline mods/d2k/maps/worm-graveyard/destroyable-tiles.yaml]: this triple whitespace is pretty weird, we don't do this. There's also some double whitespaces in this file

**Review by PunkPun — COMMENTED** [review]
I'm not sure how to feel about this feature. It's very hidden away so likely only spreadable by mouth to mouth. It also feels unpolished, should tooltips be in lower case?

Passage2

Passage1

https://github.com/OpenRA/OpenRA/assets/37534529/43b50955-6104-424e-bb8c-7ab665365a70

**Comment by Porenutak** [comment]
This feature is here -/+ year and a half. Its part of Enhanced maps and other map mods inherit it to. The reason why engi didnt dissapier is because of balance. Noone wanna spend 400$ just for cliff repair that can be destroyed again. 
Ussually this feautre is used as obsticle to capture neutral building in early game.
Snímka obrazovky (26)
Snímka obrazovky (25)

If you feel its a problem a can talk to map author if he agree to remove them. 
My English is terrible, so if someone come with better tooltip description I will include  it.

**Comment by Porenutak** [comment]
> I'm not sure how to feel about this feature. It's very hidden away so likely only spreadable by mouth to mouth. It also feels unpolished, should tooltips be in lower case?>

Maybe I should add Briefing description?

**Comment by PunkPun** [comment]
I'm not sure what's the best way.

Perhaps the fist step would be not requiring force attack to kill it? It just feels like the cliff is basically invisible and one has no way to learn about it unless explicitly taught

**Comment by Porenutak** [comment]
This feature should act same as for example bridges in RA. You dont wanna autotarget brige, you wanna past thru it or stay there or destroy it only when situation require that (same as in RA). So force force is the good way. Repaireble cliffs are shown on Radar as actors and tooltip give you more info. I dont know how to hint players more without explicit description in mission briefing (which I do when this feautre come out)

**Comment by Porenutak** [comment]
Rebased.

Worm cementery map:
- custom tiles converted from r8 - r16
- add description to engineer, that he can repair destroyed cliffs
- changed destroyed cliffs into repaired state, so even if someone didnt notice this map feature it wouldnt be a problem.

File: mods/d2k/maps/worm-graveyard/destroyable-tiles.yaml [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	Targetable:
+		TargetTypes: Ground, Structure
+		RequiresForceFire: true
+	Armor:
+		Type: wall
+	Health:
+		HP: 30000
+	Tooltip:
+		GenericName: pasage (Destroable)
+		Name: passage (Destroable)
```
Comment (PunkPun) [inline mods/d2k/maps/worm-graveyard/destroyable-tiles.yaml]:
```suggestion
		Name: Passage (destroyable)
```

these should be using our translations system, and do standardise the capital letters and fix spelling

Comment (Porenutak) [reply mods/d2k/maps/worm-graveyard/destroyable-tiles.yaml]: Do we have some dokumentation on how translations works? Can I just overide en.ftl in  custom maps?

Comment (PunkPun) [reply mods/d2k/maps/worm-graveyard/destroyable-tiles.yaml]: you can run the `--extract-yaml-strings` command. Then you can format it to match the others

Comment (PunkPun) [reply mods/d2k/maps/worm-graveyard/destroyable-tiles.yaml]: the system is currently in progress, and no, maps can't overwrite mod level translations. You need unique translation keys. The lint checks, that is `--check-yaml` should catch any errors you might have. Though the fact that this PR didn't upset the CI gods makes me suspicious that something broke

**Comment by Porenutak** [comment]
Map rules use translation now.

**Comment by PunkPun** [comment]
The discussion on discord indicates that there's a desire to have tiles defined at a game level instead of by the map

**Comment by Porenutak** [comment]
- Destroyable tiles now defined inside the mod. Assets moved into bits folder
- engineer description updated
- fixed AI attacking cliffs

**Comment by Porenutak** [comment]
removed some redundant lines in destroyable-tiles.yaml

File: mods/d2k/languages/rules/en.ftl [hunk 3]
```diff
@@ -96,8 +96,9 @@ actor-light-inf =
 
 actor-engineer =
    .description = Infiltrates and captures enemy structures
-      Strong vs Buildings
-      Weak vs Everything
+     Strong vs Buildings
+     Weak vs Everything
+     Can repair destroyed cliffs
```
Comment (PunkPun) [inline mods/d2k/languages/rules/en.ftl]: the rest of the file has 3 space intendation, while this has 2

File: mods/d2k/languages/rules/en.ftl [hunk 4]
```diff
@@ -472,3 +473,12 @@ actor-combat-tank-o =
    .encyclopedia = The Combat Tank is effective against most vehicles, less so against lightly armored vehicles.
 
     The Ordos Combat Tank is the fastest variant of the Combat Tank, but it is also the weakest.
+
+
```
Comment (PunkPun) [inline mods/d2k/languages/rules/en.ftl]: please remove double whitespace

**Review by PunkPun — DISMISSED** [review]
LGTM

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/6b1d8d2eae959705a0b482316dc8ca9230ac37a7...e4bbbcd838af1b38e2a22f5f4355a6e418390e04)

**Pull request merged**